File: WebAPI/WebAPI.Models/Basic/Project.cs
using System;

namespace WebAPI.Models.Basic
{
    public class Project
    {
        public Guid ProjectId { get; set; }
        
        public string ProjectName { get; set; }
        
        public string ProjectDescription { get; set; }
        
        public DateTime StartDate { get; set; }
        
        public DateTime EndDate { get; set; }
        
        public DateTime CreationDate { get; set; }
        
        public Guid WorkSpaceId { get; set; }
        
        public bool IsDeleted { get; set; }
    }
}

File: web/src/mappers/project.ts
import { IProject, IProjectPage, IProjectSimpleModel } from '../types/project';
import { mapToEpicModel } from './epic';
import { mapToSimpleTeamModel } from './team';

export const mapToProjectModel = (data): IProject => ({
    projectId: data.projectId,
    projectName: data.projectName,
    projectDescription: data.projectDescription,
    startDate: new Date(data.startDate),
    endDate: new Date(data.endDate),
    workSpaceId: data.workSpaceId,
    creationDate: data.creationDate,
});

export const mapToProjectSimpleModel = (data): IProjectSimpleModel => ({
    projectId: data.projectId,
    projectName: data.projectName,
    startDate: new Date(data.startDate),
    endDate: new Date(data.endDate),
});

export const mapToProjectPageModel = (data): IProjectPage => ({
    project: mapToProjectModel(data.project),
    epics: data.epics?.length ? data.epics.map(mapToEpicModel) : [],
    teams: data.teams?.length ? data.teams.map(mapToSimpleTeamModel) : [],
});
